Bottom: 968244264999205b95750ebd2b4624f85e6d332f
Top:    f5f9fcf5a9579b3d7ad0a99958577102299fb4cc
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-07-16 15:27:19 -0500

Refresh of update-ptxdist-build-factories

---
diff --git a/ansible-playbook/buildbot.yml b/ansible-playbook/buildbot.yml
index beac6956..d64e77fe 100644
--- a/ansible-playbook/buildbot.yml
+++ b/ansible-playbook/buildbot.yml
@@ -6,20 +6,20 @@
   become_user: root
   become_method: sudo
   tags:
-  - buildbot
-  pre_tasks:
+    - buildbot
+    - buildbot-master
+  tasks:
   - name: Scan Gitlab ssh keys
     command: ssh-keyscan {{gitlab_hostname}}
     register: gitlab_keys
-  roles:
   - name: Build and start the Buildbot master container
-    role: buildbot-master
-    tags:
-    - buildbot-master
+    import_role: 
+      name: buildbot-master
     vars:
       known_hosts: "{{gitlab_keys.stdout}}"
   - name: Add buildbot user to Gitlab
-    role: gitlab-user
+    import_role: 
+      name: gitlab-user
     vars:
       new_user: buildbot
       new_user_name: Build Robot
@@ -32,17 +32,29 @@
   become_user: root
   become_method: sudo
   tags:
-  - buildbot
-  - buildbot-worker
+    - buildbot
+    - buildbot-worker
   vars:
-    buildbot_master: "{{ hostvars.buildbot_master.ip_addr }}"
+    asset_host: "{{ hostvars['buildbot-worker-assets'].uri }}"
+    pypi_host: "{{ hostvars['buildbot-worker-devpi'].uri }}"
+    buildbot_master: "{{ hostvars['buildbot-master'].ip_addr }}"
+    container_networks:
+      - name: "{{ worker_backend_network }}"
+        links:
+          - buildbot-worker-assets:assets
+          - buildbot-worker-devpi:pypi
+          - buildbot-worker-ptxdist:ptxdist
     known_hosts: "{{gitlab_keys.stdout}}"
-  pre_tasks:
+  tasks:
   - name: Scan Gitlab ssh keys
     command: ssh-keyscan {{gitlab_hostname}}
     register: gitlab_keys
-  roles:
-  - role: buildbot-worker-ptxdist
+  - import_role: 
+      name: buildbot-worker-assets
+  - import_role: 
+      name: buildbot-worker-devpi
+  - import_role: 
+      name: buildbot-worker-ptxdist
 
 - name: Gitlab public keys for buildbot containers
   hosts: daedalus
@@ -50,17 +62,17 @@
   become_user: root
   become_method: sudo
   tags:
-  - buildbot
+    - buildbot
   roles:
   - name: Add buildbot_master pubkey to buildbot user
     role: gitlab-sshkey
     vars:
       username: buildbot
-      pubkey: "{{ hostvars.buildbot_master.pubkey }}"
+      pubkey: "{{ hostvars['buildbot-master'].pubkey }}"
       title: "buildbot_master"
   - name: Add buildbot_worker_ptxdist pubkey to buildbot user
     role: gitlab-sshkey
     vars:
       username: buildbot
-      pubkey: "{{ hostvars.buildbot_worker_ptxdist.pubkey }}"
+      pubkey: "{{ hostvars['buildbot-worker-ptxdist'].pubkey }}"
       title: "buildbot_worker_ptxdist"
diff --git a/roles/buildbot-worker-ptxdist/defaults/main.yml b/roles/buildbot-worker-ptxdist/defaults/main.yml
index b84c5b8b..0ed3f8d0 100644
--- a/roles/buildbot-worker-ptxdist/defaults/main.yml
+++ b/roles/buildbot-worker-ptxdist/defaults/main.yml
@@ -36,6 +36,9 @@ container_env:
 # the name of the container being started
 container_name: '{{ docker_name_prefix }}{{ hostname }}'
 
+# port configuration of the container
+container_port_args: []
+
 # volumes mounted within the container
 container_volumes:
 - '{{ data_volume }}:{{ data_path }}:z'
@@ -49,7 +52,7 @@ data_path: '/buildbot'
 data_volume: '{{ docker_name_prefix }}{{ hostname }}-data'
 
 # hostname
-hostname: buildbot-worker-ptxdist
+hostname: '{{ role_name }}'
 
 # address of hsm cryptographic service
 hsm_host: '{{ansible_facts.default_ipv4.address}}'
@@ -58,8 +61,6 @@ hsm_host: '{{ansible_facts.default_ipv4.address}}'
 image_args:
   BUILDBOT_UID: '{{buildbot_uid}}'
   BUILDBOT_VERSION: 'v{{buildbot_version}}'
-  PTK_URI: '{{asset_host}}/610-009981-015_SW_PTK_5.3_Client_RevA.tar'
-  PTK_HSM_HOST: '{{hsm_host}}'
 
 # directory of the image source
 image_dir: '{{ docker_projects_dir }}/{{ hostname }}'
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 1a45e06a..91dec6ce 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -63,6 +63,15 @@
     repo: '{{ image_repo }}'
     version: master
     dest: '{{ image_dir }}'
+    force: yes
+
+- template:
+    src: pip.conf.j2
+    dest: '{{ image_dir }}/buildbot/.pip/pip.conf'
+
+- template:
+    src: pydistutils.cfg.j2
+    dest: '{{ image_dir }}/buildbot/.pydistutils.cfg'
 
 - name: Create buildbot worker image
   docker_image:
@@ -76,10 +85,10 @@
   docker_container:
     name: '{{ container_name }}'
     image: '{{ image_name }}:{{ docker_image_tag }}'
-    # networks: '{{ buildbot_worker_ptxdist_networks }}'
+    networks: '{{ container_networks }}'
     volumes: '{{ container_volumes }}'
     env: '{{ container_env }}'
-    # ports: '{{ buildbot_worker_ptxdist_port_args }}'
+    ports: '{{ container_port_args }}'
 
 - name: Grab ssh public key
   command: >
@@ -89,5 +98,5 @@
 
 - name: Add buildbot_worker_ptxdist host
   add_host:
-    name: buildbot_worker_ptxdist
+    name: "{{ hostname }}"
     pubkey: "{{ output.stdout }}"
diff --git a/roles/buildbot-worker-ptxdist/templates/pip.conf.j2 b/roles/buildbot-worker-ptxdist/templates/pip.conf.j2
new file mode 100644
index 00000000..a54ae09a
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/templates/pip.conf.j2
@@ -0,0 +1,3 @@
+[global]
+trusted-host = {{hostvars['buildbot-worker-devpi'].uri | urlsplit('hostname')}}
+index-url = {{hostvars['buildbot-worker-devpi'].uri}}
diff --git a/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2 b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
new file mode 100644
index 00000000..f9fd2372
--- /dev/null
+++ b/roles/buildbot-worker-ptxdist/templates/pydistutils.cfg.j2
@@ -0,0 +1,2 @@
+[easy_install]
+index_url = {{hostvars['buildbot-worker-devpi'].uri}}
